test/api/search.test.ts

```diff
@@ -0,0 +1,56 @@
+import { Package } from "../../api/domain";
+import { tagsSearch } from "../../api/search";
+import { flow, pipe } from "fp-ts/function";
+import {
+  array,
+  either,
+} from "fp-ts";
+import * as A from 'fp-ts/Array'
+
+describe("search API", () => {
+  const generatePackageWithTags = (tags: Array<string>) => {
+    return {
+      "tags": tags,
+      "name": "",
+      "author": "",
+      "description": "",
+      "title": "",
+      "version": "1.2.3",
+      "archive_url": "",
+      "archive_sha256_url": ""
+    }
+  };
+
+  test("selecting multiple tags combines all possible results", () => {
+    const packagesJson = [
+      generatePackageWithTags(["a"]),
+      generatePackageWithTags(["b"]),
```

Comment:
It's already ok as the current use case is simple, but I would add a third package to the test so we are sure that `tagsSearch` is actually filtering, even with future changes. What do you think?
```suggestion
      generatePackageWithTags(["a"]),
      generatePackageWithTags(["b"]),
      generatePackageWithTags(["c"])
```

Reply:
I'm thinking we take the idea a step further, like:
- each package can have multiple tags, a random sample from A...F 
- we have 3 packages like you suggest 
- the tags to be filtered is all of A...F 
And so we still expect all packages to show

Multiple tags could mean we can also ensure the results are sorted by best match

Reply:
- Having multiple tags on each package is a good idea
- I would avoid to have the filtered array length equals to the starting one: it does not test filtering out wrong packages
- I would avoid randomly generated tags and the ones to be filtered should be a subset of the total present in all packages

In my mind this assure more robustness :) we can address the sorted matching in another PR